source/code/plugin/kubernetes_container_inventory.rb

```diff
@@ -175,17 +175,20 @@ def obtainContainerEnvironmentVars(containerId)
         unless @@containerCGroupCache.has_key?(containerId)
           $log.info("KubernetesContainerInventory::obtainContainerEnvironmentVars fetching cGroup parent pid @ #{Time.now.utc.iso8601} for containerId: #{containerId}")
           Dir["/hostfs/proc/*/cgroup"].each do |filename|
-            if File.file?(filename) && File.foreach(filename).grep(/#{containerId}/).any?
-              # file full path is /hostfs/proc/<cGroupPid>/cgroup            
-              cGroupPid = filename.split("/")[3]
-              if @@containerCGroupCache.has_key?(containerId)
-                tempCGroupPid = @@containerCGroupCache[containerId]
-                if tempCGroupPid > cGroupPid
+            begin
+              if File.file?(filename) && File.foreach(filename).grep(/#{containerId}/).any?
+                # file full path is /hostfs/proc/<cGroupPid>/cgroup
+                cGroupPid = filename.split("/")[3]
+                if @@containerCGroupCache.has_key?(containerId)
+                  tempCGroupPid = @@containerCGroupCache[containerId]
+                  if tempCGroupPid > cGroupPid
+                    @@containerCGroupCache[containerId] = cGroupPid
+                  end
+                else
                   @@containerCGroupCache[containerId] = cGroupPid
                 end
-              else
-                @@containerCGroupCache[containerId] = cGroupPid
               end
+            rescue SystemCallError # ignore Error::ENOENT,Errno::ESRCH which is expected if any of the container gone while we read
             end
```

Comment:
Can you add a log with a message or exception details rather than noop?

Reply:
> Can you add a log with a message or exception details rather than noop?

Intentionally didnt log the errors related to file errors since these will be huge if the container getting deleted.